test/functional/collection_tests.js

```diff
@@ -12,1648 +11,1054 @@ describe('Collection', function() {
+      db.createCollection('shouldCorrectlyReadBackDocumentWithNull', {}, (err, collection) => {
+        // Insert a document with a date
+        collection.insertOne({ test: null }, this.configuration.writeConcernMax(), err => {
+          expect(err).to.not.exist;
 
-          db.collection('shouldCorrectlyHandle0asIdForSave').save({ _id: 0 }, function(err) {
-            test.equal(null, err);
-            client.close();
+          collection.findOne((err, result) => {
+            expect(err).to.not.exist;
+            expect(result.test).to.not.exist;
             done();
           });
         });
       });
-    }
-  });
-
-  /**
-   * @ignore
-   */
-  it('should correctly execute update with . field in selector', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
-      });
+    });
 
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
+    /**
+     * @ignore
+     */
+    it('should throw error due to illegal update', function(done) {
+      db.createCollection('shouldThrowErrorDueToIllegalUpdate', {}, (err, coll) => {
+        try {
+          coll.update({}, null, () => {});
+        } catch (err) {
+          expect(err.message).to.equal('document must be a valid JavaScript object');
+        }
 
-        db
-          .collection('executeUpdateWithElemMatch')
-          .update({ 'item.i': 1 }, { $set: { a: 1 } }, function(err) {
-            test.equal(null, err);
+        try {
+          coll.update(null, null, () => {});
+        } catch (err) {
+          expect(err.message).to.equal('selector must be a valid JavaScript object');
+        }
 
-            client.close();
-            done();
-          });
+        done();
       });
-    }
-  });
+    });
 
-  /**
-   * @ignore
-   */
-  it('should correctly execute update with elemMatch field in selector', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
+    /**
+     * @ignore
+     */
+    it('should correctly handle 0 as id for save', function(done) {
+      db.collection('shouldCorrectlyHandle0asIdForSave').save({ _id: 0 }, err => {
+        expect(err).to.not.exist;
+
+        db.collection('shouldCorrectlyHandle0asIdForSave').save({ _id: 0 }, err => {
+          expect(err).to.not.exist;
+          done();
+        });
       });
+    });
 
-      client.connect(function(err, client) {
-        var db = client.db(self
```

Comment:
Are these tests the same as the ones in `selectorTests` with the same titles?

Reply:
They are very similar but I separated then because the `updateOne` function for the `selectorTests` uses one extra argument, and I thought it would be cleaner to separate them rather than use `if/else` statements. Should I put them all into the same test array?

Reply:
Deleted the duplicate tests in `selectorTests`